raiden-network/light-client
# Fix per channel event monitoring and past events fetching

Fixes #1434 

**Short description**
Refactors the channel monitoring logic, which became more complex than it should. With little logs to investigate, and after some real life observation of some events being lost in very specific cases, this became the main target for thsi issue.
I did test by opening ~9 channels with hub, and this issue was not experienced.

**Definition of Done**


**Steps to manually test the change (dApp)**

1. Open and close a lot of channels with other nodes, with deposits, and check if any event is lost in the process
2. It helps by reducing the default timeouts. It was discovered that hub's raiden-py still accepts transfers with channels at `{ settleTimeout: 60 }` (passed directly to `Raiden.openChannel` method), which is a huge improvement on waiting time over default=500.
